02/app.js

```diff
@@ -0,0 +1,47 @@
+document.addEventListener('DOMContentLoaded', function () {
+    const form = document.querySelector('form');
+
+    form.addEventListener('submit', function (e) {
+        e.preventDefault();
+
+        let errors = [];
+        clearErrors();
+
+        const emailInput = document.querySelector('input[name="login"]');
+        const password1Input = document.querySelector('input[name="pass1"]');
+        const password2Input = document.querySelector('input[name="pass2"]');
+        const termsCheckbox = document.querySelector('input[name="accept"]');
+        
+        switch (true) {
```

Comment:
Ciekawe podejście do `switch`, ale obawiam się, że jest ono na tyle nie standardowe, że może wprowadzać w błąd ;)
Jeśli chodzi o walidację to zdecydowanie polecam coś takiego:  https://devmentor.pl/b/walidacja-formularza-w-javascript